### ISSUE TITLE
Configuration to exclude SKIPPED tests in Allure Report

### ISSUE THREAD

**Issue by Sanjay-Bhatta** [allure-framework/allure-java#412]
#### I'm submitting a ... 
  - [ ] bug report
  - [ ] feature request
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

#### What is the current behavior?

Current behavior is Allure Tests include skip tests as well

#### If the current behavior is a bug, please provide the steps to reproduce and if possible a 

I have a total of 3 tests out of which 1 PASS, 1 FAIL & 1 SKIPPED
When I generate allure report, it says - 
Total - 3
Pass - 1
Fail - 1
Skipped - 1

Pass percentage - 33.33%

#### What is the expected behavior?

There should be a way to ignore the Skipped test cases. It will be good to have a configurable parameter to calculate the percentage. In my case I would like the pass percentage to be 50% as the Skipped test did not execute

#### Please tell us about your environment:

| Allure version | 2.8.1 |
| --- | --- |
| Test framework | testng@6.8 |
| Allure integration | allure-testng@2.0-BETA11 |
| Generate report using | allure-maven@2.18 |

#### Other information 
Please let me know if there is already a way to configure this

**Comment by pgavriluk** [comment]
A few months ago, I was looking for the same option, to remove skipped tests from the Allure statistics. Couldn't find it, so I ended writing my own statistics to display in the report email and slack channel, but the problem that if somebody clicks on the Allure report link it shows wrong numbers. So, yeah, it would be great to have it as part of the Allure report. 
Thanks.

**Comment by mszajowskiwr** [comment]
I agree, such configuration option would be a blast.

**Comment by JohnnyChiang** [comment]
Also need this config, too.

**Comment by Badhron** [comment]
yes please i need it too.

**Comment by fleytman** [comment]
Тоже очень нужна данная функция. Думал, что @Muted должен это делать, но нет. Схожие запросы есть с 2017 года. Не хочется какие-то свои велосипеды горадить.

**Comment by shridharkalagi** [comment]
Is there any workaround to solve this for the time being?

**Comment by pgavriluk** [comment]
@shridharkalagi, yes, instead of using @Disabled/@Ignore annotations (or any other annotation you might use in your framework to skip tests) I add tag "ignored" to the tests I want to ignore/skip. Like this @Tag("ignored").

Now, when you run your tests just don't run tests with that tag "ignored". 
For me, because I use mvn to run, it's: 
mvn clean test -Dgroups="{any tags you want to run, or just keep it empty}" -DexcludedGroups="ignored"

This way no ignored tests will be shown in Allure reports, but they are also won't be part of your total tests that you run. If it’s fine for you, it can be a workaround to keep Allure statistics cleaner.

**Comment by ManishIBM** [comment]
I also need this functionality

**Comment by Isidor2811** [comment]
any news here?

**Comment by anagornaia** [comment]
Yeah, we need the same functionality. On our project, we need to skip tests on condition and would be nice for them not to appear in the report as the statistic of run shows wrong numbers

**Comment by bogalx** [comment]
same here. please add this functionality or some workaround

**Comment by ManishIBM** [comment]
> Is there any workaround to solve this for the time being?

you can use hook, following is the code sample:
def pytest_collection_modifyitems(items, config):
    deselect_tcs = []
    selected_tcs = []
    for tc_item in items:
        res = check_and_disable_test(tc_item)
        if res is not None:
            deselect_tcs.append(tc_item)
        else:
            selected_tcs.append(tc_item)

    config.hook.pytest_deselected(items=deselect_tcs)
    items[:] = selected_tcs

**Comment by bogalx** [comment]
> > Is there any workaround to solve this for the time being?
> 
> you can use hook in conftest.py, following is the code sample:
> 
> def pytest_collection_modifyitems(items, config): deselect_tcs = [] selected_tcs = [] for tc_item in items: res = check_and_disable_test(tc_item) if res is not None: deselect_tcs.append(tc_item) else: selected_tcs.append(tc_item)
> 
> ```
> config.hook.pytest_deselected(items=deselect_tcs)
> items[:] = selected_tcs
> ```

Thanks but I mean in Java

**Comment by pgajeshwar-coupa** [comment]
Good to have this configuration option in allure

**Comment by easy-selenium-2022** [comment]
After spending more than two weeks and a long shell script that worked on my local runs, but failed on jenkins... this is what finally worked for me. Just two lines of code:

cd $PWD/target
grep -l status.*skipped allure-results/*result.json
grep -l status.*skipped allure-results/*result.json | xargs rm

**Note**: You need https://plugins.jenkins.io/postbuildscript/ jenkins post build and it needs to go right before Allure Report post build. Also, this is only removing the allure json result file that has the skipped status. You can still view the skipped results in the testng or junit reports.

**Comment by TimeInvestor** [comment]
We also need such a configuration.

**Comment by storenth** [comment]
Just use bash to remove skipped tests from the test results before report generation step:
```bash
grep -rl '"status": "skipped"' allure-results | xargs rm -rf
```

**Comment by romixb** [comment]
Ok, so i encountered a similar problem recently - ui tests that should be ignored with a specific configuration (e.g. specific browsers)through annotations are marked as Skipped, are displayed in TestOps report and included in the run statistics.

Using Ignore or tags was not an option because as mentioned before, one test is executed in different configs. 
I tried to use allure result modification via grep in cmd but we use withAllureUpload function in Jenkins dsl pipeline that wraps the testrun and sends results on the go. 
After a bit of debugging i found a solution that suites our case - cancel the default behaviour of `createFakeResult(final ITestContext context, final ITestNGMethod method)` from `AllureTestNg` listener by overiridong it with an empty method
```
@Override
 protected void createFakeResult(ITestContext context, ITestNGMethod method) {
}
```
This will affect those cases when skipping tests is allowed in your test run.

**Comment by QingqinLi** [comment]
I need this functionality too

**Comment by MrKozhin** [comment]
Need this config badly!

**Comment by wazzeps** [comment]
Also need this config.

**Comment by baev** [comment]
BTW, there is an option for TestNG integration to hide disabled tests from the report. See https://allurereport.org/docs/testng-configuration/#alluretestnghidedisabledtests

From what I see, there are multiple different requests from the community:

1. To exclude skipped tests in statistics. This is a language-agnostic report feature. 
2. Hide disabled tests from TestNG — there is a configuration option; see the link above.
3. Some other "hide skipped" requests from other languages — please create a separate issue/discussion for that

I suggest only discussing the number one within the scope of the issue.

**Comment by baev** [comment]
After internal discussion, we decided to change the behaviour unconditionally and exclude `unknown` and `skipped` statuses from the passed rate calculation. I'll try to include this change in the next release of the Allure Report (aiming to make it this year).

In case of positive feedback from the community, we'll propagate the change to Allure TestOps as well.

**Comment by baev** [comment]
Since this is a front-end-only change, please, try to clear the browser cache and check again. 

For me it working just fine:

<img width="769" alt="Screenshot 2024-01-08 at 14 40 42" src="https://github.com/allure-framework/allure-java/assets/2149631/54aa76b4-05fb-4058-a054-5748ae374ee0">

**Comment by wazzeps** [comment]
Could anyone confirm that clearing the cache helps?

I tried clearing the cache / opening the page in incognito, but disabled tests continue to be taken into account.

**Comment by KeaganGonsalvez** [comment]
it doesn't work for me I cleared the cache but it still accounts for skipped tests.

**Comment by baev** [comment]
Could you please check https://baev.github.io/testng-java-maven/5/# ? It shows 100% for me, ignoring unknown & skipped statuses in the success rate calculation (source code https://github.com/baev/testng-java-maven )

**Comment by wazzeps** [comment]
> Could you please check https://baev.github.io/testng-java-maven/5/# ? It shows 100% for me, ignoring unknown & skipped statuses in the success rate calculation (source code https://github.com/baev/testng-java-maven )

Can confirm it shows 100%.

**Comment by wazzeps** [comment]
I see you declared Allure Gradle plugin version is 2.12.

Could you check it with the latest released version (2.11.2)?

**Comment by wazzeps** [comment]
After some research I managed to find the source of the problem.

Allure Gradle plugin 2.11.2 with default confifuration generates a report with disabled statuses.
While calling `allure generate` manually creates the correct report.

@baev can we configure AGP to fix this?

**Comment by baev** [comment]
> can we configure AGP to fix this?

```kts
allure {
    version.set("2.25.0")
}
```

**Comment by adenzhong** [comment]
![image](https://github.com/allure-framework/allure-java/assets/113604623/3a364074-fc21-44aa-8163-0c0fbd390efc)
It works as expected, thanks.
